index.d.ts

```diff
@@ -0,0 +1,71 @@
+import EventEmitter from 'bare-events'
+
+export class Socket extends EventEmitter {
```

Comment:
As of `bare-events` v2.5.2 the `EventEmitter` class supports an event map so let's typed the events as well. For example:

```ts
const e = new EventEmitter<{ foo: [string, number] }>()

e.on('foo', (str: string, n: number) => {})
e.emit('foo', 'hello', 42)
```

Reply:
🧙